### PR TITLE
[IMP] udes_stock: Ensure last reserved pallet is cleared on drop-off

### PR DESCRIPTION
**Pull request by LoKKeR96**
Fixes the bug where the pallet is asked to be scanned again after
being dropped off in workflows.

User-story: 1827

Signed-off-by: Lorenzo Cucurachi <lorenzo.cucurachi@unipart.io>

### CODE REVIEW

File: addons/udes_stock/models/stock_picking_batch.py
Comment (armandcela) [inline addons/udes_stock/models/stock_picking_batch.py:834]: Is the update inside the loop?

**Comment by armandcela** [comment]
I think the docstring should be updated to mention that at the end u_last_reserved_pallet_name will be False

File: addons/udes_stock/models/stock_picking_batch.py
Comment (LoKKeR96) [reply addons/udes_stock/models/stock_picking_batch.py:834]: Yep, should be outside to be correct, fixing up soon.

**Comment by jrbirch99** [comment]
Q: In the story it mentions two MRs one branching off of master, and the other off of tag 11.14.2. Should there be another PR for the tagged version?

**Comment by LoKKeR96** [comment]
> Q: In the story it mentions two MRs one branching off of master, and the other off of tag 11.14.2. Should there be another PR for the tagged version?

Yes, I wanted to have a word with Adam first, also making sure code is reviewed before pushing another MR and maintaining two which is painful.

**Review by PeterAlabaster — APPROVED** [review]
LGTM once loop comment addressed

**Review by armandcela — APPROVED** [review]

**Comment by jrbirch99** [comment]
> > Q: In the story it mentions two MRs one branching off of master, and the other off of tag 11.14.2. Should there be another PR for the tagged version?
> 
> Yes, I wanted to have a word with Adam first, also making sure code is reviewed before pushing another MR and maintaining two which is painful.

Okay, but its the same fix, right?

**Comment by LoKKeR96** [comment]
> > > Q: In the story it mentions two MRs one branching off of master, and the other off of tag 11.14.2. Should there be another PR for the tagged version?
> > 
> > 
> > Yes, I wanted to have a word with Adam first, also making sure code is reviewed before pushing another MR and maintaining two which is painful.
> 
> Okay, but its the same fix, right?

Yes, the story is written a bit ambiguously and it's not clear what they mean with tagging.
A TL will tag master once this story is finished and merged.

**Review by jrbirch99 — APPROVED** [review]

**Pull request merged**
